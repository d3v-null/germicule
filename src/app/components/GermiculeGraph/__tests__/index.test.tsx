import React from 'react';
import { render } from '@testing-library/react';
import echarts from 'echarts';
import { GermiculeGraph, deconstructGermicule } from '..';

import {
  GermiculeItem,
  // GermiculeCluster,
  // GermiculeMeta,
  // GraphNode,
  // GraphEdge,
  // GraphCategory,
  GraphInfo,
} from '../../../types';

const emptyGermicule: Array<GermiculeItem> = [];
const emptyGraphInfo: GraphInfo = {
  nodes: [],
};

const unknownGermicule: Array<GermiculeItem> = [null as GermiculeItem];
const unknownGraphInfo: GraphInfo = {
  nodes: [
    {
      name: 'unknown 0',
      _label: '‚ùì',
    },
  ],
};

const lonelyGermicule: Array<GermiculeItem> = [
  {
    name: 'ü¶Ñ',
    risk: 5,
  } as GermiculeItem,
];
const lonelyGraphInfo: GraphInfo = {
  nodes: [
    {
      name: 'ü¶Ñ',
      _label: 'ü¶Ñ',
      value: 5,
    },
  ],
};

const twinGermicule: Array<GermiculeItem> = [
  {
    name: 'üåè',
    risk: 3,
    germicule: [
      {
        name: 'üåö',
        risk: 2,
        contact: 5,
        description: 'sattelite',
      } as GermiculeItem,
    ],
  },
];
const twinGraphInfo: GraphInfo = {
  nodes: [
    {
      name: 'üåè',
      _label: 'üåè',
    },
    {
      name: 'üåö',
      _label: 'üåö',
    },
  ],
  edges: [
    {
      source: 'üåè',
      target: 'üåö',
      value: 5,
      _label: 'sattelite',
    },
  ],
};

let spy: any;

beforeAll(() => {
  spy = jest.spyOn(echarts, 'getInstanceByDom').mockImplementation(() => {
    return {
      hideLoading: jest.fn(),
      setOption: jest.fn(),
      showLoading: jest.fn(),
    };
  });
});
afterAll(() => {
  spy.mockRestore();
});

describe('<GermiculeGraph  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(
      <GermiculeGraph data={{ germicules: lonelyGermicule }} />,
    );
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});

describe('deconstructGermicule', () => {
  it('should handle empty germicule data', () => {
    const result = deconstructGermicule(emptyGermicule);
    expect(result).toMatchObject(emptyGraphInfo);
  });
  it('should handle unknown germicule data', () => {
    const result = deconstructGermicule(unknownGermicule);
    expect(result).toMatchObject(unknownGraphInfo);
  });
  it('should handle lonely germicule data', () => {
    const result = deconstructGermicule(lonelyGermicule);
    expect(result).toMatchObject(lonelyGraphInfo);
  });
  it('should handle twin germicule data', () => {
    const result = deconstructGermicule(twinGermicule);
    expect(result).toMatchObject(twinGraphInfo);
  });
});
