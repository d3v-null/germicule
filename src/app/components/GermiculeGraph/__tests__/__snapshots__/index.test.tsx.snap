// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GermiculeGraph  /> should match snapshot 1`] = `
.c0 {
  width: 'auto';
  height: '50vh';
  margin: 'auto';
}

<div
  class="c0"
  id="germicule-graph"
>
  <div
    _echarts_instance_="ec_1598149906409"
    class="echarts-for-react "
    size-sensor-id="1"
    style="height: 300px; user-select: none;"
  >
    <div
      style="position: relative; width: 0px; height: 300px; padding: 0px; margin: 0px; border-width: 0px;"
    />
    <object
      class="size-sensor-object"
      data="about:blank"
      style="display: block; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1; opacity: 0;"
      tabindex="-1"
      type="text/html"
    />
  </div>
</div>
`;
